pkg/cmd/autoscaling/internal/targetlifecyclestate/parse.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package targetlifecyclestate
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/autoscaling/targetlifecyclestate"
+)
+
+type (
+	targetLifecycleStateSchedule struct {
+		Initial     targetlifecyclestate.TargetState
+		Transitions []scheduledTransition
+	}
+
+	scheduledTransition struct {
+		State targetlifecyclestate.TargetState
+		Time  time.Time
+	}
+)
+
+// parse parses the target lifecycle state transitions from the given arguments.
+// Args should be empty or have an odd length. Every even indexed item must be
+// a valid lifecycle target state, odd indexed items must be a RFC3339 time or
+// duration.
+func parse(args []string) (p targetLifecycleStateSchedule, err error) {
+	if len(args) == 0 {
+		return p, nil
+	}
+
+	p.Initial, err = targetlifecyclestate.Parse(args[0])
+	if err != nil {
+		return p, fmt.Errorf("parsing initial state: %w", err)
+	}
+
+	args = args[1:]
```

Comment:
[nit]: I think keeping args in the original form will keep the logic below more intuitive and close to the comments above " odd indexed items must be a RFC3339 time or duration".  Was there a specific reason for the reassignment - looks like the reasoning was to think of args as scheduledTransitions beyond this point?